app/controllers/api/v1/search_results_controller.rb

```diff
@@ -4,10 +4,7 @@ module Api
   module V1
     class SearchResultsController < ApplicationController
       def create
-        upload_form = UploadForm.new(
-          search_engine: upload_form_params[:search_engine],
-          csv_file: upload_form_params[:csv_file]
-        )
+        upload_form = create_upload_form
```

Comment:
This code is not idiomatic Ruby. A more idiomatic Ruby would be using the double pipe equal operator (read more about it: https://medium.com/@tali.scheer/ruby-or-equals-operator-6aed32af2a66):

```ruby

def create
  if upload_form.valid?
    #...
  end
end

private

def upload_form
  @upload_form ||= UploadForm.new(
    search_engine: upload_form_params[:search_engine],
    csv_file: upload_form_params[:csv_file]
  )
end 
```
 
Ruby has this motto: "Ask, Don’t Tell" (see https://www.cloudbees.com/blog/use-ask-dont-tell-policy-ruby), which aims to make code more elegant.